OCR.py

```diff
@@ -0,0 +1,162 @@
+            if np.abs(.5*pt1down-.5*pt1up)>50:#(np.sum(im[index,0,:]) > 80) and (np.sum(im[index-1,0,:]) > 80):
+                #index+=1
+                break
+            index-=1
+            
+        metadatastrip = im[index:,:,:]
+        return metadatastrip
+    if camera_type == 'hpwren':
+        index = 10
+        xview =10
+        
+        while 30>index:
+            pt1up = np.sum(im[index-1,:xview,:])
+            pt1down =np.sum(im[index+1,:xview,:]) 
+            if np.abs(.5*pt1down-.5*pt1up)>160*xview:#(np.sum(im[index,:xview,:]) <1000) and (np.sum(im[index+1,:xview,:]) <1000):
+                index=math.ceil(index*1.5)#index+=3#add a buffer for lower than average chars like g,j,q,p...
+                break
+            index+=1
+        metadatastrip = im[:index,:,:]
+        return metadatastrip
+    return None
+  
+def test_iden(filename,cam_type):
+    """test function to assess the capability of the metadata location and cropping
+    Args:
+        filename (str) : filepath
+        camera_type (str): {'hpwren','Axis','unknown'} defined type of image to remove metadata from.
+    Returns:
+        None
+    """
+    img = load_image( filename )
+    metadata = cut_metadata(img,cam_type)
+    save_image(metadata,"test4.jpg")
+    tic=time.time()
+    logging.warning('time taken to cut metadata %s',time.time()-tic)
+
+
+
+
+
+
+
+
+
+
+
+import pytesseract
+import time
+def ocr_core(filename=None, data=None):
+    """
+    This function will handle the core OCR processing of images.
+    Args:
+        opt filename (str) : filepath
+        opt data (array): data
+    Returns:
+        list of OCR recognized data
+    """
+    if filename:
+        text = pytesseract.image_to_string(load_image( filename ),config="-c tessedit_char_whitelist=%s_-." % char_whitelist)
+    if type(data) == np.ndarray:
+        text = pytesseract.image_to_string(data,config="-c tessedit_char_whitelist=%s_-." % char_whitelist)
+    else:
+        logging.warning('Please feed in processable data to ocr_core of type filename or data')
+        return
+    return text.split(' ')
+
+
+
+def pull_metadata(camera_type,filename = None, data = None, save_location=False):
+    """test function to assess the capability of the OCR
+    Args:
+        opt filename (str) : filepath
+        opt data (array): data
+        camera_type (str): {'hpwren','Axis','unknown'} defined type of image to remove metadata from.
+        opt save_location (str): filepath to save metadata strip to
+    Returns:
+        vals (list): list of OCR recognized data
+    """
+    if filename:
+        img = load_image( filename )
+    elif type(data) == np.ndarray:
+        img = data
+    else:
+        logging.warning('specify data location or data itself')
+    metadata = cut_metadata(img,camera_type)
+    tic=time.time()#####################
+    vals = ocr_core(data = metadata)
+    logging.warning('time to complete OCR: %s',time.time()-tic)##################
```

Comment:
As mentioned earlier, would be good to know the breakdown between OCR vs. cropping.